prow/tide/history/history.go

```diff
@@ -91,6 +195,22 @@ func (h *History) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Flush writes the action history to persistent storage if configured to do so.
+func (h *History) Flush() {
```

Comment:
Can we just export a `map[string][]Record`? This seems easier and more portable across versions.

Reply:
That is pretty much what I had before, but I switched to storing the `History.logs` field instead since human readability isn't a concern: https://github.com/kubernetes/test-infra/pull/11016#discussion_r252466548

Switching to storing a `map[string][]Record` would still require a "smell" like the one you noted above in order to reconstruct the recordLogs from the `[]Record`s and it would be less efficient (always rebuilding the recordLogs instead of only when the record limit changes and reversing and copying the records on every flush).

Reply:
> always rebuilding the recordLogs instead of only when the record limit changes and reversing and copying the records on every flush

I suspect the cost of this is dwarfed by the cost of the reflection and building the a string of json-encoded bytes and writing them.

Reply:
Switched to storing `map[string][]Record` as requested.